File: Medium/Number of pairs/number-of-pairs.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;


// } Driver Code Ends
 

class Solution{
    public:
    
    // X[], Y[]: input array
    // M, N: size of arrays X[] and Y[] respectively
    //Function to count number of pairs such that x^y is greater than y^x.
    long long countPairs(int X[], int Y[], int M, int N)
    {
       sort(X,X+M);
        sort(Y,Y+N);
        int one=0;
        for(int i=0;i<N;i++) if(Y[i]==1) one++;
        long long ans=0;
        for(int i=0;i<M;i++){
            int x=X[i];
            if(x==1) continue;
            if(x==2){
                int ind=upper_bound(Y,Y+N,4)-Y;
                ans+=N-ind+one;
            }
            else if(x==3){
                int ind=upper_bound(Y,Y+N,x)-Y;
                ans+=N-ind;
                ind=upper_bound(Y,Y+N,2)-Y;
                ans+=ind;
            }
            else{
                int ind=upper_bound(Y,Y+N,x)-Y;
                ans+=N-ind+one;
            }
        }
        return ans;
    }
};


//{ Driver Code Starts.
int main()
{
	int T;
	cin>>T;
	while(T--)
	{
		int M,N;
		cin>>M>>N;
		int i,a[M],b[N];
		for(i=0;i<M;i++)
		{
			cin>>a[i];
		}
		for(i=0;i<N;i++)
		{
			cin>>b[i];
		}
		Solution ob;
		cout<<ob.countPairs(a, b, M, N)<<endl;
	}
}
// } Driver Code Ends

File: Difficulty: Medium/Number of pairs/number-of-pairs.py
#User function Template for python3

import math
class Solution:
    
     #Function to count number of pairs such that x^y is greater than y^x.     
    def countPairs(self,arr,brr):
        for i in range(len(arr)):
            arr[i]=math.log(arr[i])/arr[i]
        for i in range(len(brr)):
            brr[i]=math.log(brr[i])/brr[i]
        arr.sort()
        brr.sort()
        cnt=0
        for i in range(len(arr)):
            idx=bisect.bisect_left(brr,arr[i])
            cnt+=idx
        return cnt


#{ 
 # Driver Code Starts
#Initial Template for Python 3

#Initial Template for Python 3

import atexit
import io
import sys
import bisect

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER


@atexit.register
def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())


if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        # M, N = map(int, input().strip().split())
        a = list(map(int, input().strip().split()))
        b = list(map(int, input().strip().split()))
        ob = Solution()
        print(ob.countPairs(a, b))
        #code here

# } Driver Code Ends